pkg/codegen/hcl2/model/type_table.go

```diff
@@ -0,0 +1,140 @@
+package model
+
+type typeTable struct {
```

Comment:
Ha, so we need this because go does not let us build hashmaps with custom hash/equality? That seems to be true, I googled for libraries, but there is nothing out there, how sad.

Reply:
That's correct. Waiting for generics on that one...

Reply:
Sadness.. the universe is cold and unforgiving.